crates/jstz_cli/src/repl.rs

```diff
@@ -5,41 +5,98 @@ use jstz_api::{
     encoding::EncodingApi, http::HttpApi, js_log::set_js_logger, url::UrlApi,
     urlpattern::UrlPatternApi, ConsoleApi, KvApi,
 };
-use jstz_core::host::HostRuntime;
 use jstz_core::{
+    host::HostRuntime,
     host_defined,
     kv::Kv,
     runtime::{self, Runtime},
 };
 use jstz_proto::api::{ContractApi, LedgerApi};
-use rustyline::{error::ReadlineError, Editor};
+use rustyline::{
+    completion::Completer, error::ReadlineError, highlight::Highlighter, hint::Hinter,
+    validate::Validator, Editor, Helper,
+};
+use std::borrow::Cow;
 use tezos_smart_rollup_mock::MockHost;
 
 use syntect::{
     easy::HighlightLines,
     highlighting::{Style, Theme, ThemeSet},
     parsing::{SyntaxReference, SyntaxSet},
-    util::LinesWithEndings,
-};
-
-use crossterm::{
-    cursor::MoveUp,
-    terminal::{Clear, ClearType},
-    ExecutableCommand,
 };
-use std::io::{stdout, Write};
 
 use crate::{config::Config, debug_api::DebugApi};
 
-pub fn exec(self_address: Option<String>, cfg: &Config) -> Result<()> {
-    // Initialize syntax set and theme set
-    let ss = SyntaxSet::load_defaults_newlines();
-    let ts = ThemeSet::load_defaults();
+struct JsHighlighter {
+    ss: SyntaxSet,
+    syntax: SyntaxReference,
+    theme: Theme,
+}
+
+impl JsHighlighter {
+    pub fn new() -> Self {
+        // Initialize syntax set and theme set only once
+        let ss = SyntaxSet::load_defaults_newlines();
+        let ts = ThemeSet::load_defaults();
+
+        // Find JavaScript syntax and a theme
+        let syntax = ss.find_syntax_by_extension("js").unwrap();
+        let theme = &ts.themes["base16-ocean.dark"];
+
+        JsHighlighter {
+            ss: ss.clone(),
+            syntax: syntax.clone(),
+            theme: theme.clone(),
+        }
```

Comment:
Possibly unnecessary `.clone`s

Reply:
It seems to actually require clone, as ss/syntax/them should not be references.

Reply:
Indeed, having looked at the `syntect` docs, I don't think the clones for `syntax` and `theme` can be removed. However `ss.clone()` can be removed.

```suggestion
        JsHighlighter {
            syntax: syntax.clone(),
            theme: theme.clone(),
            ss,
        }
```